ompi/mpi/c/request_free.c

```diff
@@ -57,6 +57,10 @@ int MPI_Request_free(MPI_Request *request)
     }
 
     rc = ompi_request_free(request);
+    if (OPAL_UNLIKELY(OMPI_SUCCESS != rc && MPI_REQUEST_NULL != *request)) {
+        (*request)->req_status.MPI_ERROR = rc;
```

Comment:
Did `ompi_request_free(request)` already set this?

Reply:
This is the generic user-level  `MPI_Request_free()` for all kinds of requests, not only generalized requests. I'm not so expert with the Open MPI implementation to 100% sure  `ompi_request_free()` will set the status error field on errors. Without that guarantee, my use of `ompi_errhandler_request_invoke()` may not work as I intended. Note that `ompi_errhandler_request_invoke()` has the side-effect of cleaning up the request and deallocating the object. Sorry, it's a bit complicated, but I had to work within the constraints of what I was given.

For example, look at `ompi_comm_request_free()`... I don't see that routine setting the status error field. Therefore, I think being overzealous here is justified. IMHO, this line is a good one and should stay. Otherwise, all internal request types used in Open MPI (and future ones that may be implemented) should be audited and eventually fixed to make sure a failing free sets the error field.